charts/newrelic-infrastructure/templates/ksm/deployment.yaml

```diff
@@ -117,6 +117,12 @@ spec:
                   apiVersion: "v1"
                   fieldPath: "spec.nodeName"
 
+            - name: "NRIA_OVERRIDE_HOSTNAME"
+              valueFrom:
+                fieldRef:
+                  apiVersion: "v1"
+                  fieldPath: "spec.nodeName"
+
```

Comment:
Do we need these on the non-kubelet pods? They are running with `IS_FORWARD_ONLY`

Reply:
Yep. We had a GTS ticket about kube-events also hammering the DNS: https://github.com/newrelic/helm-charts/pull/670

So this change should also be done in all the other charts @paologallinaharbur

Reply:
I am elaborating a bit further. 

For pods that are not sending the hostname because they are in forwarding mode, it may not make sense because it does not send information about the hostname although they are hammering the DNS server, and as [Ruben pointed out](https://github.com/newrelic/nri-kubernetes/pull/492#discussion_r896916188) "If both short and full are overridden, then no DNS resolution will be made, so I would stick to these."

So we are not solving only the issue of hammering DNS by putting a good "short hostname" but also by not making any DNS query.

Reply:
Yes it is **not** needed to avoid to send the fullHostname. 

However, I prefer to add it as kang-makes mentioned to:
 - avoid useless DNS queries
 - to be sure to have the exact same behaviour of hostnames across the three components.

Reply:
If we want to avoid hammering the DNS, I'd rather pass `DnsHostnameResolution=false` than overriding the hostname.

My main concern is that the agent in these pods should not report any hostname altogether. If we set this to the same name with which other agents _will_ be reporting, we might hide bugs, or confuse ourselves later.

Reply:
If the hostname is overridden (`override_hostname`) then the DNS will not be queried

Reply:
Then I change it to leverage `NRIA_DNS_HOSTNAME_RESOLUTION` for KSM and controlPlane